core/src/main/kotlin/riven/core/service/connotation/ConnotationAdminService.kt

```
@@ -0,0 +1,87 @@
+     */
+    @PreAuthorize("@workspaceSecurity.hasWorkspace(#workspaceId)")
+    fun reanalyzeWhereVersionMismatch(
+        metadataType: ConnotationMetadataType,
+        tier: AnalysisTier,
+        workspaceId: UUID,
+    ): Int {
+        require(metadataType == ConnotationMetadataType.SENTIMENT) {
+            "Phase B only supports SENTIMENT reanalyze (got $metadataType)"
+        }
+        if (tier != AnalysisTier.DETERMINISTIC) {
+            throw NotImplementedError("Phase B only supports DETERMINISTIC reanalyze (got $tier)")
```

detekt (1.23.8)

[warning] The NotImplementedDeclaration should only be used when a method stub is necessary. This defers the development of the functionality of this function. Hence, the `NotImplementedDeclaration` should only serve as a temporary declaration. Before releasing, this type of declaration should be removed.

(detekt.exceptions.NotImplementedDeclaration)